src/common/meta/src/key/runtime_switch.rs

```diff
@@ -0,0 +1,226 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+use std::time::Duration;
+
+use common_error::ext::BoxedError;
+use common_procedure::local::PauseAware;
+use moka::future::Cache;
+use snafu::ResultExt;
+
+use crate::error::{GetCacheSnafu, Result};
+use crate::key::{LEGACY_MAINTENANCE_KEY, MAINTENANCE_KEY, PAUSE_PROCEDURE_KEY};
+use crate::kv_backend::KvBackendRef;
+use crate::rpc::store::{BatchDeleteRequest, PutRequest};
+
+pub type RuntimeSwitchManagerRef = Arc<RuntimeSwitchManager>;
+
+/// The runtime switch manager.
+///
+/// Used to enable or disable runtime switches.
+#[derive(Clone)]
+pub struct RuntimeSwitchManager {
+    kv_backend: KvBackendRef,
+    cache: Cache<Vec<u8>, Option<Vec<u8>>>,
+}
+
+#[async_trait::async_trait]
+impl PauseAware for RuntimeSwitchManager {
+    async fn is_paused(&self) -> std::result::Result<bool, BoxedError> {
+        self.is_procedure_paused().await.map_err(BoxedError::new)
+    }
+}
+
+const CACHE_TTL: Duration = Duration::from_secs(10);
+const MAX_CAPACITY: u64 = 32;
+
+impl RuntimeSwitchManager {
+    pub fn new(kv_backend: KvBackendRef) -> Self {
+        let cache = Cache::builder()
+            .time_to_live(CACHE_TTL)
+            .max_capacity(MAX_CAPACITY)
+            .build();
+        Self { kv_backend, cache }
+    }
+
+    async fn put_key(&self, key: &str) -> Result<()> {
+        let req = PutRequest {
+            key: Vec::from(key),
+            value: vec![],
+            prev_kv: false,
+        };
+        self.kv_backend.put(req).await?;
+        self.cache.invalidate(key.as_bytes()).await;
+        Ok(())
+    }
+
+    async fn delete_keys(&self, keys: &[&str]) -> Result<()> {
+        let mut req = BatchDeleteRequest::default();
+        for key in keys {
+            req = req.add_key(key.as_bytes());
+        }
```

Comment:
to avoid unnecessary vec allocation:

```suggestion
        let req = BatchDeleteRequest::new()
            .with_keys(keys.iter().map(|x| x.as_bytes().to_vec()).collect())
```